queasars/minimum_eigensolvers/base/termination_criteria.py

```diff
@@ -42,34 +42,48 @@ class BestIndividualChangeTolerance(EvolvingAnsatzMinimumEigensolverBaseTerminat
     :param minimum_change: absolute expectation value improvement below which the algorithm shall terminate.
         Must be at least 0
     :type minimum_change: float
+    :param allowed_consecutive_violations: determines how often the threshold value can be violated consecutively
+        before this criterion chooses to terminate. If set to 0, this terminates the first time the change falls
+        below the threshold value. If set to 2 for example, this terminates the first time the change falls below
+        the threshold three consecutive times. Must be at least 0.
+    :type allowed_consecutive_violations: int
     """
 
-    def __init__(self, minimum_change: float):
+    def __init__(self, minimum_change: float, allowed_consecutive_violations: int = 0):
         if minimum_change <= 0:
             raise ValueError("The minimum absolute improvement parameter must be bigger than 0!")
+        if allowed_consecutive_violations < 0:
+            raise ValueError("allowed_consecutive_violations must be at least 0!")
 
         self._minimum_change: float = minimum_change
-        self._previous_expectation_value: float = float("inf")
+        self._allowed_consecutive_violations: int = allowed_consecutive_violations
+        self._previous_expectation_value: Optional[float] = None
+        self._change_history: list[float] = []
 
     def reset_state(self) -> None:
-        self._previous_expectation_value = float("inf")
+        self._change_history = []
```

Comment:
Shouldn't this also reset `self._previous_expectation_value`? It is used in line 77 for calculating the next `change`, right?

Reply:
Correct. It seems i missed adding it to reset_state(). I corrected this accordingly.